### DISCUSSION TITLE
Some Issues

### DISCUSSION THREAD

**Discussion by andrew-secret** [OpenSourceRaidGuild/website#8] [Ideas]
Hey there 👋,

I took a look into the repo of the website and my first impression was that we can already create some issues around it regarding scaffolding and project structure before we finish the wireframes.

1. Maybe we can already implement Continous Deployment with testing etc w/ Netlify
2. Another Issue I had in my mind is to implement Storybook. This could help us build encapsulated UI w/o having so much merge conflicts and we got a better overview about our application
3. Would about adding a general Code coverage to the project?
4. Deploy the App and a URL to the repo in the description

Another thing I recognized that we add firebase credentials to the GitHub. Is this intended?

You can leave some feedback/ideas in the thread ✌️❤️

**Comment by tigerabrodi** [comment]
@andrew-secret Good points! :fire:

**Comment by marcosvega91** [comment]
Hello guys, I think that continuous deployment on netlify will be awesome, you can also enable the deploy on PRs. If there is no much traffic it will be free

**Reply by andrew-secret** [reply]
Sounds good! Are there any other opinions regarding this topic (continuous deployment)?
I would love to create an issue for that. Do you want to join me @marcosvega91 ?

**Reply by marcosvega91** [reply]
Yes of course, the issue could be good so we can track it. Let's wait for the other's member thoughts

**Comment by nobrayner** [comment]
1. Were you thinking of setting this up via GitHub actions? Or does Netlify itself have that built in - I've never used it! If all you need is for someone to create a netlify account and drop a secret into the repo, go for it! Just let me know what to call the secret :stuck_out_tongue: 

2. Storybook was definitely on my mind, though I have never used it. I'm all for getting it set up.

3. Can you elaborate more on "general" code coverage?

4. Once 1. is done, this will definitely happen :sunglasses:

**Reply by tigerabrodi** [reply]
I think Code Coverage is something we can add when we start adding tests or afterward, the goal would be to make sure it does not drop drastically I guess, though what is most important is to cover the behaviors of a user :fire:

**Reply by marcosvega91** [reply]
Yes netlify has a github integration that we can configure. It has a limit of deployment in months but I think that should be good for this repo

**Reply by andrew-secret** [reply]
> 1. Were you thinking of setting this up via GitHub actions? Or does Netlify itself have that built in - I've never used it! If all you need is for someone to create a netlify account and drop a secret into the repo, go for it! Just let me know what to call the secret 😛
> 2. Storybook was definitely on my mind, though I have never used it. I'm all for getting it set up.
> 3. Can you elaborate more on "general" code coverage?
> 4. Once 1. is done, this will definitely happen 😎

@nobrayner Yeah sure I thought of having a code coverage for our frontend and cover this with unit tests. For instance I know this tool here https://about.codecov.io/
One trade-off could be that the contribution barrier is increasing because we have to deal with testing. And I wouldn't set the codecov to 100% I would suggest something like **80%** is totally fine. I would like to avoid the case that people are writing tests only for coverage purposes

Regarding storybook: I would create an issue today and add more information to it, if it's okay.

**Reply by nobrayner** [reply]
Go for it! More than happy to see storybook setup!

**Reply by andrew-secret** [reply]
I've created an issue next step I'll wait till people add some reactions or feedback. Then I will fork the repo and try to implement it :)
https://github.com/OpenSourceRaidGuild/website/issues/13

**Reply by tigerabrodi** [reply]
@andrew-secret Sounds nice! :fire: 

As for code coverage, in my opinion, it can come later, not urgent. :man_shrugging:. What is most important to me, is that we cover those behaviors. :100:

**Comment by tigerabrodi** [comment]
@marcosvega91 Do you have any experience with Vercel?

To me, it seems like it is more worth than Netlify. https://vercel.com/guides/deploying-svelte-with-vercel, from what I have heard, it is cheaper and more flexible than Netlify.

I think we should give Vercel a try before Netlify :fire: 

"After your project has been imported, all subsequent pushes to branches will generate Preview Deployments, and all changes made to the Production Branch (commonly "main") will result in a Production Deployment.". - Vercel

Just two baby steps to deploy it! :raised_hands:

**Reply by tigerabrodi** [reply]
Already been discussed and decided here https://discord.com/channels/715220730605731931/785649901782433852/801537353163341844
